From f912db3f8dc0c0cb8cb2358e6ec79dd03b7a875d Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Mon, 8 Dec 2025 09:28:05 +0100
Subject: [PATCH] util/driconf/asahi: Override GL renderer for web browsers

Several web sites block clients with "Apple" in the WebGL renderer
string if the reported OS is not one of Apple's.
This check seems to implemented via a 3rd party product which is slowly
rolled out over more web sites. Instead of playing whack-a-mole with
web sites in multiple browsers override the OpenGL renderer in mesa for
known browsers.

Backport-to: 25.3
Signed-off-by: Janne Grunau <j@jannau.net>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/38845>
---
 src/util/00-mesa-defaults.conf | 45 ++++++++++++++++++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/src/util/00-mesa-defaults.conf b/src/util/00-mesa-defaults.conf
index 70108e6892fb5..d4ba00727c7b9 100644
--- a/src/util/00-mesa-defaults.conf
+++ b/src/util/00-mesa-defaults.conf
@@ -1372,6 +1372,51 @@ TODO: document the other workarounds.
         </application>
     </device>

+    <device driver="asahi">
+        <!-- Override GL renderer for known web browsers since one or more
+             web security frameworks apparently block clients with "Apple"
+             in the the renderer string if the OS is not one of Apple's.
+             Affected web sites include paypal.com, etsy.com and wsj.com.
+             Reported as web-compat bug in
+             https://github.com/webcompat/web-bugs/issues/189524
+        -->
+        <application name="Firefox" executable="firefox">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Firefox ESR" executable="firefox-esr">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Chromium" executable="chromium">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Chromium" executable="chromium-browser">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Google Chrome" executable="chrome">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="vivaldi-bin" executable="vivaldi-bin">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Iceweasel" executable="iceweasel">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Epiphany" executable="epiphany">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Konqueror" executable="konqueror">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Falkon" executable="falkon">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Seamonkey" executable="seamonkey">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+        <application name="Waterfox" executable="waterfox">
+            <option name="force_gl_renderer" value="AGX G13/G14"/>
+        </application>
+    </device>
     <device driver="hk">
         <engine engine_name_match="DXVK|vkd3d">
            <!--
--
GitLab
